AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Migration/MutationSyncMetadataMigrationDelegate+SQLiteValidation.swift

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import SQLite
+import AWSPluginsCore
+
+extension SQLiteMutationSyncMetadataMigrationDelegate {
+
+    // MARK: - Precondition
+
+    /// Migration requires some hardcoded SQL statements that may break if the schema changes
+    /// Ensure that the schemas used in this migration are as expected
+    func preconditionCheck() throws {
+        guard MutationSyncMetadata.schema.fields.count == 4 else {
+            throw DataStoreError.internalOperation("MutationSyncMetadata schema has changed from 4 fields", "", nil)
+        }
+
+        guard MutationSyncMetadata.schema.fields["id"] != nil,
+              MutationSyncMetadata.schema.fields["deleted"] != nil,
+              MutationSyncMetadata.schema.fields["lastChangedAt"] != nil,
+              MutationSyncMetadata.schema.fields["version"] != nil else {
+            throw DataStoreError.internalOperation("MutationSyncMetadata schema missing expected fields", "", nil)
+        }
+
+        guard MutationSyncMetadataCopy.schema.fields.count == 4 else {
+            throw DataStoreError.internalOperation("MutationSyncMetadataCopy schema has changed from 4 fields", "", nil)
+        }
+
+        guard MutationSyncMetadataCopy.schema.fields["id"] != nil,
+              MutationSyncMetadataCopy.schema.fields["deleted"] != nil,
+              MutationSyncMetadataCopy.schema.fields["lastChangedAt"] != nil,
+              MutationSyncMetadataCopy.schema.fields["version"] != nil else {
+            throw DataStoreError.internalOperation("MutationSyncMetadataCopy schema missing expected fields", "", nil)
+        }
```

Comment:
nit: would checking that `MutationSyncMetadata` fields and `MutationSyncMetadataCopy` fields are exactly the same achieve the same result here?

Reply:
i think the current check is more simple as they are explicit string checks that aren't referencing their actual field names. If we check the fields are exactly the same, even though it would be rare, but both structs could change at the same time and then this would pass (but should have failed). we could keep the explicit field count == 4 but still run into an even more rare scenario where the fields were changed in both models and count is still 4.

Reply:
i pulled out the strings as local variables to the this method as a way to at least reuse them for both models. let me know if you prefer a different check or if i misinterpreted what you meant by checking MutationSyncMetadata fields and MutationSyncMetadataCopy fields are exactly the same

Reply:
makes sense! I thought we only cared about `MutationSyncMetadata` and `MutationSyncMetadataCopy` being in sync